File: Code/app/src/main/java/com/bigfriedicecream/recipes/interfaces/IRecipeContract.java
package com.bigfriedicecream.recipes.interfaces;

import com.bigfriedicecream.recipes.models.RecipeDataModel;

public interface IRecipeContract {

    interface View {
        void render(RecipeDataModel recipe);
    }

    interface Presenter {
        void start();
        void stop();
        void load(String id);
    }
}

File: Code/app/src/main/java/com/bigfriedicecream/recipes/interfaces/IRecipeContract.kt
package com.bigfriedicecream.recipes.interfaces

import com.bigfriedicecream.recipes.models.RecipeDataModel

interface IRecipeContract {

    interface View {
        fun render(recipe: RecipeDataModel)
    }

    interface Presenter {
        fun start()
        fun stop()
        fun load(id:String)
    }
}
